src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -971,16 +963,264 @@ void CmdSketcherViewSection::activated(int iMsg)
+                auto checked = checkbox->checkState() == Qt::Checked;
+
+                if( value != checked ) {
+                    const QSignalBlocker blocker(checkbox);
+                    checkbox->setChecked(value);
+                }
+            };
+
+            auto updateCheckBoxFromProperty = [updateCheckBox](QCheckBox * checkbox, App::PropertyBool & property) {
+                auto propvalue = property.getValue();
+
+                updateCheckBox(checkbox, propvalue);
+            };
+
+            updateCheckBox(gridSnap, sketchView->getSnapMode() == SnapMode::SnapToGrid);
+
+            updateCheckBoxFromProperty(gridAutoSpacing, sketchView->GridAuto);
+
+            auto autospacing = gridAutoSpacing->checkState() == Qt::Checked;
+
+            gridSizeBox->setEnabled(!autospacing);
+
+            gridSizeBox->setValue(sketchView->GridSize.getValue());
+        }
+    }
+
+    void languageChange()
+    {
+        gridSnap->setText(QApplication::translate("GridSpaceAction", "Grid Snap"));
+        gridSnap->setToolTip(QApplication::translate("GridSpaceAction", "New points will snap to the nearest grid line.\nPoints must be set closer than a fifth of the grid spacing to a grid line to snap."));
+        gridSnap->setStatusTip(gridSnap->toolTip());
+
+        gridAutoSpacing->setText(QApplication::translate("GridSpaceAction", "Grid Auto Spacing"));
+        gridAutoSpacing->setToolTip(QApplication::translate("GridSpaceAction", "Resize grid automatically depending on zoom."));
+        gridAutoSpacing->setStatusTip(gridAutoSpacing->toolTip());
+
+        sizeLabel->setText(QApplication::translate("GridSpaceAction", "Spacing"));
+        gridSizeBox->setToolTip(QApplication::translate("GridSpaceAction", "Distance between two subsequent grid lines"));
+    }
+
+protected:
+    QWidget* createWidget(QWidget* parent) override
+    {
+        gridSnap = new QCheckBox(QApplication::translate("GridSpaceAction", "Grid Snap"));
+        gridSnap->setToolTip(QApplication::translate("GridSpaceAction", "New points will snap to the nearest grid line.\nPoints must be set closer than a fifth of the grid spacing to a grid line to snap."));
+        gridSnap->setStatusTip(gridSnap->toolTip());
+
+        gridAutoSpacing = new QCheckBox(QApplication::translate("GridSpaceAction", "Grid Auto Spacing"));
+        gridAutoSpacing->setToolTip(QApplication::translate("GridSpaceAction", "Resize grid automatically depending on zoom."));
+        gridAutoSpacing->setStatusTip(gridAutoSpacing->toolTip());
+
+        sizeLabel = new QLabel(QApplication::translate("GridSpaceAction", "Spacing"));
+        gridSizeBox = new Gui::QuantitySpinBox();
+        gridSizeBox->setProperty("unit", QVariant(QStringLiteral("mm")));
+        gridSizeBox->setObjectName(QStringLiteral("gridSize"));
+        gridSizeBox->setToolTip(QApplication::translate("GridSpaceAction", "Distance between two subsequent grid lines"));

```

Comment:
IMO best to apply here untranslated texts (or even empty strings) then call `languageChange()` at the end.

Reply:
Yes. I removed all the string from the creation and reused the languageChange.